papis/docmatcher.py

```diff
@@ -186,8 +214,57 @@ def get_regex_from_search(search: str) -> Pattern[str]:
         properly escaped and allows for multiple spaces.
     """
     return re.compile(
-        ".*{}.*".format(".*".join(map(re.escape, search.split()))),
-        re.IGNORECASE)
+        ".*{}.*".format(".*".join(map(re.escape, search.split()))), re.IGNORECASE
+    )
+
+
+def check_syntax(parsed: List[ParseResult]) -> bool:
+    """Tests the syntax by replacing all search terms with True,
+    then trying to evaluate the resulting string.
+    """
+
+    test = []
+    for p in parsed:
+        if not p.syntax:
+            test.append("True")
+        else:
+            test.append(p.string)
+
+    try:
+        eval(" ".join(test))
+        return True
+    except SyntaxError:
+        logger.error("Malformed query.")
+        return False
+
+
+def proofread(parsed: List[ParseResult]) -> List[ParseResult]:
```

Comment:
This could benefit from a more obvious name. Perhaps `add_implicit_query_operators` or something? I'm not good at naming :(

Reply:
You're right proofread is rather abstruse. I will use your suggestion or something similar!